core-primitives/attestation-handler/src/cert.rs

```diff
@@ -276,62 +282,71 @@ where
-		},
+	let mut payload = cert_der[offset..offset + len].to_vec();
+	println!("payload in mra cert verifier is: {:#?}", &payload);
+	if is_payload_base64_encoded {
+		payload = base64::decode(&payload[..]).or(Err(sgx_status_t::SGX_ERROR_UNEXPECTED))?;
 	}
+	println!("payload in mra cert verifier is: {:#?}", &payload);
+	if !is_dcap {
+		// Extract each field
+		let mut iter = payload.split(|x| *x == b'|');
+		let attn_report_raw = iter.next().ok_or(sgx_status_t::SGX_ERROR_UNEXPECTED)?;
+		let sig_raw = iter.next().ok_or(sgx_status_t::SGX_ERROR_UNEXPECTED)?;
+		let sig = base64::decode(sig_raw).map_err(|e| EnclaveError::Other(e.into()))?;
+
+		let sig_cert_raw = iter.next().ok_or(sgx_status_t::SGX_ERROR_UNEXPECTED)?;
+		let sig_cert_dec = base64::decode_config(sig_cert_raw, base64::STANDARD)
+			.map_err(|e| EnclaveError::Other(e.into()))?;
+		let sig_cert = webpki::EndEntityCert::from(&sig_cert_dec).expect("Bad DER");
+
+		// Verify if the signing cert is issued by Intel CA
+		let mut ias_ca_stripped = IAS_REPORT_CA.to_vec();
+		ias_ca_stripped.retain(|&x| x != b'\r' && x != b'\n');
+		let head_len = "-----BEGIN CERTIFICATE-----".len();
+		let tail_len = "-----END CERTIFICATE-----".len();
+		let full_len = ias_ca_stripped.len();
+		let ias_ca_core: &[u8] = &ias_ca_stripped[head_len..full_len - tail_len];
+		let ias_cert_dec = base64::decode_config(ias_ca_core, base64::STANDARD)
+			.map_err(|e| EnclaveError::Other(e.into()))?;
+
+		let mut ca_reader = BufReader::new(IAS_REPORT_CA);
+
+		let mut root_store = rustls::RootCertStore::empty();
+		root_store.add_pem_file(&mut ca_reader).expect("Failed to add CA");
+
+		let trust_anchors: Vec<webpki::TrustAnchor> =
+			root_store.roots.iter().map(|cert| cert.to_trust_anchor()).collect();
+
+		let now_func = webpki::Time::try_from(SystemTime::now());
+
+		match sig_cert.verify_is_valid_tls_server_cert(
+			SUPPORTED_SIG_ALGS,
+			&webpki::TLSServerTrustAnchors(&trust_anchors),
+			&[ias_cert_dec.as_slice()],
+			now_func.map_err(|_e| EnclaveError::Time)?,
+		) {
+			Ok(_) => info!("Cert is good"),
+			Err(e) => {
+				error!("Cert verification error {:?}", e);
+				return Err(sgx_status_t::SGX_ERROR_UNEXPECTED)
+			},
+		}
 
-	// Verify the signature against the signing cert
-	match sig_cert.verify_signature(&webpki::RSA_PKCS1_2048_8192_SHA256, attn_report_raw, &sig) {
-		Ok(_) => info!("Signature good"),
-		Err(e) => {
-			error!("Signature verification error {:?}", e);
-			return Err(sgx_status_t::SGX_ERROR_UNEXPECTED)
-		},
-	}
+		// Verify the signature against the signing cert
+		match sig_cert.verify_signature(&webpki::RSA_PKCS1_2048_8192_SHA256, attn_report_raw, &sig)
+		{
+			Ok(_) => info!("Signature good"),
+			Err(e) => {
+				error!("Signature verification error {:?}", e);
+				return Err(sgx_status_t::SGX_ERROR_UNEXPECTED)
+			},
+		}
 
-	verify_attn_report(attn_report_raw, pub_k, attestation_ocall)
+		verify_attn_report(attn_report_raw, pub_k, attestation_ocall)
+	} else {
+		Ok(())
+	}
```

Comment:
Ah todo with an issue here would be nice.

Reply:
Great idea :100: , added in [21d1756](https://github.com/integritee-network/worker/pull/1375/commits/21d175653b3644b189f28134a0953d428457369f).